pkg/minishift/openshift/service.go

```diff
@@ -195,35 +199,13 @@ func calculateWeight(a float64, b float64) string {
 func filterAndUpdateServiceSpecs(routeSpec map[string][]ServiceWeight, namespace string, serviceNodePort map[string]string) []ServiceSpec {
 	var serviceSpecs []ServiceSpec
 	serviceSpecMap := make(map[string]*ServiceSpec)
-
 	for routeURL, v := range routeSpec {
 		for _, service := range v {
-			// split content on colon to separate NAME, HOST and Weight
 			if _, ok := serviceNodePort[service.Name]; ok {
-				if _, ok := serviceSpecMap[service.Name]; !ok {
-					serviceSpecMap[service.Name] = &ServiceSpec{Namespace: namespace,
-						Name:     service.Name,
-						URL:      []string{fmt.Sprintf("http://%s", routeURL)},
-						NodePort: serviceNodePort[service.Name],
-						Weight:   []string{service.Weight},
-					}
-				} else {
-					serviceSpecMap[service.Name].URL = append(serviceSpecMap[service.Name].URL, fmt.Sprintf("http://%s", routeURL))
-					serviceSpecMap[service.Name].Weight = append(serviceSpecMap[service.Name].Weight, service.Weight)
-				}
+				addStuffToServiceSpecMap(serviceSpecMap, routeURL, service, namespace, serviceNodePort[service.Name])
```

Comment:
why named `Stuff`?

Reply:
Not able to find a better name.. i suck at naming, suggestions are welcome!

Reply:
Ohh, did you mean just `addToServiceSpecMap`? :-)